src/Functions/MultiMatchAllIndicesImpl.h

```diff
@@ -120,6 +121,112 @@ struct MultiMatchAllIndicesImpl
             offset = haystack_offsets[i];
         }
 #else
+        (void)haystack_data;
+        (void)haystack_offsets;
+        (void)needles_arr;
+        (void)res;
+        (void)offsets;
+        (void)edit_distance;
+        (void)max_hyperscan_regexp_length;
+        (void)max_hyperscan_regexp_total_length;
+        throw Exception(
+            "multi-search all indices is not implemented when vectorscan is off",
+            ErrorCodes::NOT_IMPLEMENTED);
+#endif // USE_VECTORSCAN
+    }
+
+    static void vectorVector(
+        const ColumnString::Chars & haystack_data,
+        const ColumnString::Offsets & haystack_offsets,
+        const ColumnArray & needles_col,
+        PaddedPODArray<ResultType> & res,
+        PaddedPODArray<UInt64> & offsets,
+        bool allow_hyperscan,
+        size_t max_hyperscan_regexp_length,
+        size_t max_hyperscan_regexp_total_length)
+    {
+        vectorVector(haystack_data, haystack_offsets, needles_col, res, offsets, std::nullopt, allow_hyperscan, max_hyperscan_regexp_length, max_hyperscan_regexp_total_length);
+    }
+
+    static void vectorVector(
+        const ColumnString::Chars & haystack_data,
+        const ColumnString::Offsets & haystack_offsets,
+        const ColumnArray & needles_col,
+        PaddedPODArray<ResultType> & res,
+        PaddedPODArray<UInt64> & offsets,
+        std::optional<UInt32> edit_distance,
+        bool allow_hyperscan,
+        size_t max_hyperscan_regexp_length,
+        size_t max_hyperscan_regexp_total_length)
+    {
+        if (!allow_hyperscan)
+            throw Exception(ErrorCodes::FUNCTION_NOT_ALLOWED, "Hyperscan functions are disabled, because setting 'allow_hyperscan' is set to 0");
+#if USE_VECTORSCAN
+        size_t prev_haystack_offset = 0;
+        for (size_t i = 0; i < haystack_offsets.size(); ++i)
+        {
+            Field field = needles_col[i];
+            const Array & needles_arr = DB::get<Array &>(field);
+
+            std::vector<std::string_view> needles;
+            needles.reserve(needles_arr.size());
+            for (const auto & needle : needles_arr)
+                needles.emplace_back(needle.get<String>());
```

Comment:
It doesn't look effective, we copy each array from column to array field. We can avoid copying using nested column and offsets from array column to create an array of string views (actually our `StringRef` from `getDataAt` IColumn method)

Reply:
True, it's not effective :D Let me try to come up with something more intelligent.